File: app/src/main/java/org/digitalcampus/oppia/adapter/DevicesBTAdapter.java
package org.digitalcampus.oppia.adapter;


import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import org.digitalcampus.mobile.learning.R;

import java.util.List;

public class DevicesBTAdapter extends RecyclerViewClickableAdapter<DevicesBTAdapter.DevicesBTViewHolder> {

    private List<String> devices;
    private Context context;


    public DevicesBTAdapter(Context context, List<String> devices) {
        this.context = context;
        this.devices = devices;
    }

    @Override
    public DevicesBTViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        View contactView = LayoutInflater.from(context).inflate(R.layout.row_device, parent, false);

        // Return a new holder instance
        return new DevicesBTViewHolder(contactView);
    }


    @Override
    public void onBindViewHolder(final DevicesBTViewHolder holder, final int position) {

        final String device = getItemAtPosition(position);

        holder.rootView.setText(device);

    }

    @Override
    public int getItemCount() {
        return devices.size();
    }

    public String getItemAtPosition(int position) {
        return devices.get(position);
    }


    public class DevicesBTViewHolder extends RecyclerViewClickableAdapter.ViewHolder {

        private TextView rootView;

        public DevicesBTViewHolder(View itemView) {

            super(itemView);

            rootView = (TextView) itemView;
        }

    }

}

File: app/src/main/java/org/digitalcampus/oppia/adapter/DevicesBTAdapter.kt
package org.digitalcampus.oppia.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import org.digitalcampus.mobile.learning.R
import org.digitalcampus.mobile.learning.databinding.RowDeviceBinding
import org.digitalcampus.oppia.adapter.DevicesBTAdapter.DevicesBTViewHolder

class DevicesBTAdapter(private val context: Context, private val devices: List<String>) : RecyclerViewClickableAdapter<DevicesBTViewHolder>() {

    inner class DevicesBTViewHolder(itemView: View) : ViewHolder(itemView) {
        val binding: RowDeviceBinding

        init {
            binding = RowDeviceBinding.bind(itemView)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DevicesBTViewHolder {
        val contactView = LayoutInflater.from(context).inflate(R.layout.row_device, parent, false)
        return DevicesBTViewHolder(contactView)
    }

    override fun onBindViewHolder(holder: DevicesBTViewHolder, position: Int) {
        val device = getItemAtPosition(position)
        holder.binding.root.text = device
    }

    override fun getItemCount(): Int {
        return devices.size
    }

    fun getItemAtPosition(position: Int): String {
        return devices[position]
    }
}